### PR TITLE
chore(BarChart): Added tests for BarChart component

### PR DESCRIPTION
**Pull request by siggysamson**
🤷‍♂️ 

I think the `BarChart` component really should have some tests as we are adding more and more functionality to it.
It was really hard to figure out how many tests to write and how extensive they should be but here we are 😄 

As I was writing tests I:
1) Implemented a custom propTypes validation for `lower` as I didn't want to bother writing tests for what happens when `lower` is smaller than `upper` and
2) I separated the zoom calculation into `utils` so that I can write tests for it separately.

I guess some tests are still better than no tests but it was really hard to figure out what's worth writing a test for. I actually might have got lost and forgot to test some properties so if you find anything while reviewing let me know

### CODE REVIEW

File: packages/axiom-charts/src/BarChart/BarChart.js [hunk 1]
```diff
@@ -53,7 +53,19 @@ export default class BarChart extends Component {
...
+      if (!props.lower) {
+        return;
+      }
+
+      if (typeof props.lower !== 'number') {
+        return new Error('Invalid prop `lower` supplied to ' + componentName + '. Validation failed.');
+      }
+
+      if (props.upper && props.upper < props.lower) {
+        return new Error('Invalid prop `lower` supplied to ' + componentName + '. `lower` needs to be smaller than `upper`');
```
Comment (HHogg) [inline packages/axiom-charts/src/BarChart/BarChart.js]: Should PropTypes be used for value validation and how far should it be taken is a pretty hot/common topic in React 😄 There are quite a few cases in Axiom and in general react components where value validation with custom propTypes would be useful to stop silly mistakes, but it adds even more maintenance, where those mistakes actually exist in the implements. 

I'm really of the opinion that propTypes should be kept for type validation, and the components should expect they are being used in the correct way, but it's an opinion and open to others. What's your thoughts @thatguynamedandy?

For this specific use case, maybe we could add a helpful swap check instead so lower and upper are the correct way around?

File: packages/axiom-charts/src/BarChart/barChartFixture.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+export const getXAxisLabels = () => ['0', '10', '20', '30', '40', '50', '60', '70', '80', '90', '100'];
```
Comment (HHogg) [inline packages/axiom-charts/src/BarChart/barChartFixture.js]: The build process currently only excludes test files form the final published packages, so with how this current stands this fixture file would end up on NPM. 

I guess there is two options:
* Could move these fixtures into the test final ([similar to the DotPlot](https://github.com/BrandwatchLtd/axiom/blob/master/packages/axiom-charts/src/DotPlotChart/utils.test.js#L19))
* Change the build process to exclude `.fixture.js` files?

File: packages/axiom-charts/src/BarChart/utils.test.js [hunk 3]
```diff
@@ -133,4 +133,112 @@ describe('BarChart (utils)', () => {
...
+          finalLower: 5,
+        })).toBe(33.33333333333333);
+      });
+
+      test('uses dataUpper as the base for the zoom calculation when dataUpper is larger than finalUpper', () => {
+        expect(calculateZoom({
+          dataUpper: 20,
+          finalUpper: 10,
+          finalLower: 5,
+        })).toBe(300);
```
Comment (HHogg) [inline packages/axiom-charts/src/BarChart/utils.test.js]: I haven't dipped into the zooming stuff in a while but I thought what was produced was always a percentage? _(0-100)_

Comment (HHogg) [reply packages/axiom-charts/src/BarChart/utils.test.js]: I'm not actually sure, a lot of these tests are needed or the util needs to change a bit... currently the Charts are performing the min/max checks so dataUpper and finalUpper values will never be in this situation. Maybe the min/max checks need moving to the util and instead it takes the data bounds and the explicit bounds from the props? /shrug

**Comment by lpoulter** [comment]
@siggysamson whats the status of this PR?

**Comment by siggysamson** [comment]
I didn't have time to finish this up and I'd prefer to just close it and I'll reopen again when I get to it!

**Comment by siggysamson** [comment]
Thanks for the reminder about this PR 🙂

**Pull request closed without merging**